src/components/Shared/AmountInput/AmountInput.tsx

```diff
@@ -14,33 +13,75 @@ export type Props = {
 
 const AmountInput: FC<Props> = ({ amount, register, errorMessage }) => {
   const { t } = useTranslation();
+  const [amountInput, setAmountInput] = useState<string>(
+    amount ? "" + amount : ""
+  );
+  const { unit, toggleUnit } = useContext(AppContext);
+
+  const { onChange } = register;
 
-  const ButtonToggleUnit: FC = () => {
-    const { unit, toggleUnit } = useContext(AppContext);
+  const toggleHandler = () => {
+    let formattedValue = amountInput;
+    if (unit === Unit.BTC) {
+      formattedValue = new Intl.NumberFormat("en-US").format(
+        convertBtcToSat(+formattedValue)!
+      );
+      setAmountInput(formattedValue);
+    } else {
+      formattedValue = formattedValue.replace(/,|\./g, "");
```

Comment:
Wanna add a comment what this reges does?

Reply:
Done :+1: